### PR TITLE
feat: add context manager support for `bentoml.client`

### PR DESCRIPTION
**Pull request by y1450**
- implments `__enter__`,`__exit__`,`__aenter__`and `__aexit__`

A dummy implementation is added to support to context protocol as it is required for the 'with' statement.

The implementation of exit method is empty as client does not have any persistent connection, each api call is processed in a separate aio.http session.

## What does this PR address?

Fixes #(issue)

## Before submitting:

- [x] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [x] Does the code follow BentoML's code style, both `make format` and `make lint` script have passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [x] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

**Comment by y1450** [comment]
I could not find any existing test cases for initial PR (#3028).
One can add tests for clean up and setup, However going through the existing code, I could not infer what exactly need to be cleanup. 
My first idea was HTTP connection, but it is already managed through session which is created on each call. 
https://github.com/bentoml/BentoML/blob/931dedcd1fb1d5a054898236898c5dde4ac54588/src/bentoml/client.py#L163

As a testcase I tried this but I am not sure whether it is good test?
``` python
from bentoml.client import Client
import numpy as np
client = Client.from_url("http://localhost:3000")
client.classify(np.array([[4.9, 3.0, 1.4, 0.2]]))
with Client.from_url("http://localhost:3000") as cl:
    print(cl)
# should this give exception or not?    
# currently, it executes without any exception.
cl.classify(np.array([[4.9, 3.0, 1.4, 0.2]]))  
```

**Review by aarnphm — COMMENTED** [review]
With one minor comment. Otherwise, it is pretty straightforward.

Right now, we don't have a test for our client, but ideally, the client should be used for our tests feature, which is under construction 🚧 from #2968.

I'm also thinking of moving this to #3280 as I refactor and add gRPC support for `bentoml.client`.

File: src/bentoml/client.py [hunk 1]
```diff
@@ -82,6 +82,18 @@ def wait_until_server_is_ready(host: str, port: int, timeout: int) -> None:
...
+        return self
+
+    def __exit__(self, type, value, traceback):
+        pass
+
+    async def __aenter__(self):
+        return self
+
+    async def __aexit__(self, type, value, traceback):
+        pass
```
[the remark this answers is not in the document]
Comment (y1450) [reply src/bentoml/client.py:95]: This is my first review and I am new to GH ecosystem. I highly appreciate the feedback. Next time I will consider typing information :smile:  .

Should I move it to your branch #3280 ? by creating a pull request on your clone? is that the preferred way to move commit?
should I squash the 2 commits? 

I am also fine,  if you want to close this. move the code by yourself. Although I would  like to practice the workflow.

**Review by aarnphm — APPROVED** [review]
Thanks for the contritbutions.

**Pull request merged**